src/server/Services/GatewayService.cs

```diff
@@ -92,138 +61,95 @@ public async Task<IEnumerable<Store>> GetKnownStoresWithNames(CancellationToken
         try
         {
-            var cacheKey = $"{storeId}-{hexKey}-proof";
-            var proof = await _memoryCache.GetOrCreateAsync(cacheKey, async entry =>
-            {
-                entry.SlidingExpiration = TimeSpan.FromMinutes(15);
-                _logger.LogInformation("Getting proof for {StoreId} {Key}", storeId.SanitizeForLog(), hexKey.SanitizeForLog());
-
-                var fsCacheValue = await _fileCache.GetValueAsync(cacheKey, cancellationToken);
-                if (fsCacheValue is not null)
+            var proof = await _cacheService.GetOrCreateAsync($"{storeId}-{hexKey}-proof",
+                TimeSpan.FromMinutes(15),
+                async () =>
                 {
-                    _logger.LogInformation("Got value for {StoreId} {Key} from file cache", storeId.SanitizeForLog(), hexKey.SanitizeForLog());
-                    return fsCacheValue;
-                }
-
-                var proofResponse = await _dataLayer.GetProof(storeId, new List<string> { HttpUtility.UrlDecode(hexKey) }, cancellationToken);
-                var proof = proofResponse.ToJson();
-                await _fileCache.SetValueAsync(cacheKey, proof, cancellationToken);
-                _logger.LogInformation("Got proof for {StoreId} {proof} from DataLayer", storeId.SanitizeForLog(), proof.SanitizeForLog());
-                return proof;
-            });
+                    _logger.LogInformation("Getting proof for {StoreId} {Key}", storeId.SanitizeForLog(), hexKey.SanitizeForLog());
+                    var proofResponse = await _dataLayer.GetProof(storeId, [HttpUtility.UrlDecode(hexKey)], cancellationToken);
+                    return proofResponse.ToJson();
+                },
+                cancellationToken);
 
             return proof;
         }
-        catch
-        {
-            return null; // 404 in the api
-        }
-        finally
+        catch (Exception e)
         {
-            await _storeUpdateNotifierService.RegisterStoreAsync(storeId);
+            _logger.LogError(e, "Failed to get proof for {StoreId}", storeId.SanitizeForLog());
         }
+
+        return null; // 404 in the api
     }
 
     public async Task<string?> GetValue(string storeId, string key, string? rootHash, CancellationToken cancellationToken)
     {
         try
         {
-            var cacheKey = $"{storeId}-{key}";
-            var value = await _memoryCache.GetOrCreateAsync(cacheKey, async entry =>
-            {
-                entry.SlidingExpiration = TimeSpan.FromMinutes(15);
-                _logger.LogInformation("Getting value for {StoreId} {Key}", storeId.SanitizeForLog(), key.SanitizeForLog());
-
-                var fsCacheValue = await _fileCache.GetValueAsync(cacheKey, cancellationToken);
-                if (fsCacheValue is not null)
+            await _storeCacheService.VerifyStoreCache(storeId, cancellationToken);
+            
+            var value = await _cacheService.GetOrCreateAsync($"{storeId}-{key}",
```

Comment:
Im wondering if we should include the roothash in the cache key. That way we are sure the cached value belongs to the root we are looking for (in almost all cases, the latest root)

Reply:
done